cedar-policy-validator/src/schema/namespace_def.rs

```diff
@@ -132,12 +132,31 @@ impl ValidatorNamespaceDef<ConditionalName, ConditionalName> {
 
         // Convert the common types, actions and entity types from the schema
         // file into the representation used by the validator.
-        let common_types =
-            CommonTypeDefs::from_raw_common_types(namespace_def.common_types, namespace.as_ref())?;
-        let actions =
-            ActionsDef::from_raw_actions(namespace_def.actions, namespace.as_ref(), extensions)?;
-        let entity_types =
-            EntityTypesDef::from_raw_entity_types(namespace_def.entity_types, namespace.as_ref())?;
+        let common_types = CommonTypeDefs::from_raw_common_types(
```

Comment:
Could we update these `from_raw_*` functions to take an `impl IntoIterator<..>` to avoid rebuilding the map?

Reply:
Maybe another PR? This one is already huge.

Reply:
that'd be fine. I excepted this change to be trivial, but if it's not we can do it later

Reply:
This is indeed a large PR.  It probably could have been split into (1) the refactoring changes in Core and then (2) the changes in cedar-policy-validator.  But at this point I don't think it's worth making the split, just a comment for the future that splitting large PRs like this would be helpful

Reply:
The changes are indeed trivial. Made them.